### ISSUE TITLE
A problem that could be solved using CVXOPT in v1.0.31 fails in v1.1.6

### ISSUE THREAD

**Issue by jeffrey-hokanson** [cvxpy/cvxpy#1147]
**Describe the bug**
I suspect there has been a regression in the reduction for CVXOPT.  The example below runs without error in CVXPY v1.0.31, but fails in CVXPY v1.1.6.

**To Reproduce**
```
import numpy as np
import cvxpy as cp

M = 2
m = 3
X = np.random.randn(M, m)

alpha = cp.Variable(M, nonneg = True)
beta = cp.Variable()
obj = cp.Maximize(beta)
x = cp.Parameter(m)
p = cp.Parameter(m)

constraints = [ X.T @ alpha == beta * p + x, cp.sum(alpha) == 1,]
prob = cp.Problem(obj, constraints)

x.value = np.mean(X, axis = 0)
p.value = X[0] - x.value
prob.solve(solver = 'CVXOPT', verbose =True, warm_start = True)
print("alpha", alpha.value)
print("beta", beta.value)

assert np.all(np.isclose(alpha.value,np.eye(M)[:,0]))
assert np.isclose(beta.value, 1)
```

Note, this will succeed if switching to SCS or if `M`>=  `m` in both CVXPY v1.0.31 or v1.1.6.  

**Expected behavior**
This code should run and pass the assertion statements. 

**Output**

In CVXPY v1.0.31
```
     pcost       dcost       gap    pres   dres   k/t
 0:  3.8781e-16  5.5511e-16  2e+00  0e+00  3e+00  1e+00
 1: -8.2427e-01 -8.2427e-01  2e-01  3e-16  3e-01  9e-02
 2: -9.9824e-01 -9.9824e-01  2e-03  1e-16  3e-03  1e-03
 3: -9.9998e-01 -9.9998e-01  2e-05  2e-16  3e-05  1e-05
 4: -1.0000e+00 -1.0000e+00  2e-07  3e-16  3e-07  1e-07
 5: -1.0000e+00 -1.0000e+00  2e-09  7e-17  3e-09  1e-09
Optimal solution found.
alpha [9.99999999e-01 8.78634363e-10]
beta 0.9999999982427316

```

In CVXPY v1.1.6

```
Traceback (most recent call last):
  File "break_cvxpy.py", line 24, in <module>
    prob.solve(solver = 'CVXOPT', verbose =True, warm_start = True)
  File "/Users/jeffreyh/Library/Python/3.7/lib/python/site-packages/cvxpy/problems/problem.py", line 396, in solve
    return solve_func(self, *args, **kwargs)
  File "/Users/jeffreyh/Library/Python/3.7/lib/python/site-packages/cvxpy/problems/problem.py", line 754, in _solve
    self.unpack_results(solution, solving_chain, inverse_data)
  File "/Users/jeffreyh/Library/Python/3.7/lib/python/site-packages/cvxpy/problems/problem.py", line 1068, in unpack_results
    "Try another solver, or solve with verbose=True for more "
cvxpy.error.SolverError: Solver 'CVXOPT' failed. Try another solver, or solve with verbose=True for more information.
```

**Version**
 - OS: MacOS 10.15.6
 - CVXPY Version: 1.1.6
 - Python Version: 3.7.9
 - CVXOPT Version: 1.2.5

**Additional context**
In my own [unit tests](https://github.com/jeffrey-hokanson/PSDR/blob/master/tests/domains/test_convexhull.py) solving a similar system with the same dimensions, I see the error

```
        if h.size[0] != cdim:
>           raise TypeError("'h' must be a 'd' matrix of size (%d,1)" %cdim)
E           TypeError: 'h' must be a 'd' matrix of size (2,1)

../../../../Library/Python/3.7/lib/python/site-packages/cvxopt/coneprog.py:517: TypeError

```
whereas on entry, `h = <1x1 matrix, tc='d'>`.  I have been unable to make this exception occur in my MWE, but this leads me to suspect that some constraints aren't being properly concatenated.

**Comment by SteveDiamond** [comment]
You can look at the inputs to CVXOPT by calling ``prob.get_problem_data('CVXOPT')``. Then you can compare the 1.0.31 and 1.1.6 formulations.

**Comment by jeffrey-hokanson** [comment]
Looking at the problem data, it seems that in v1.1.6,  G is not populated and has the wrong shape

In v1.0.31

```
({'A': <4x3 sparse matrix of type '<class 'numpy.float64'>'
    with 11 stored elements in Compressed Sparse Column format>,
  'G': <2x3 sparse matrix of type '<class 'numpy.float64'>'
    with 2 stored elements in Compressed Sparse Column format>,
  'b': array([2.00247277e+00, 1.13385760e+00, 7.30052115e-04, 1.00000000e+00]),
  'c': array([-1.,  0.,  0.]),
  'dims': (zero: 4, nonpos: 2, exp: 0, soc: [], psd: []),
  'h': array([-0., -0.]),
  'offset': array([0.])},
 Chain(reductions=[<cvxpy.reductions.flip_objective.FlipObjective object at 0x133cc5b50>, <cvxpy.reductions.dcp2cone.dcp2cone.Dcp2Cone object at 0x134d5e690>, <cvxpy.reductions.cvx_attr2constr.CvxAttr2Constr object at 0x134d5e6d0>, <cvxpy.reductions.eval_params.EvalParams object at 0x134d5e950>, <cvxpy.reductions.dcp2cone.cone_matrix_stuffing.ConeMatrixStuffing object at 0x134d5e8d0>, <cvxpy.reductions.solvers.conic_solvers.cvxopt_conif.CVXOPT object at 0x134136d90>]),
 [[],
  <cvxpy.reductions.inverse_data.InverseData object at 0x134d5e890>,
  ({0: Variable((2,)), 1: Variable(())},
   {0: Variable((2,), nonneg=True), 1: Variable(())},
   {10: 10, 15: 15}),
  [],
  <cvxpy.reductions.inverse_data.InverseData object at 0x134d5ed90>,
  {'eq_constr': [Zero(Expression(AFFINE, UNKNOWN, (3,))),
                 Zero(Expression(AFFINE, UNKNOWN, ()))],
   'offset': 0.0,
   'other_constr': [NonPos(Expression(AFFINE, UNKNOWN, (2,)))],
   'var_id': 52}])
```

In v1.1.6

```({'A': <4x3 sparse matrix of type '<class 'numpy.float64'>'
    with 11 stored elements in Compressed Sparse Column format>,
  'G': <1x3 sparse matrix of type '<class 'numpy.float64'>'
    with 0 stored elements in Compressed Sparse Column format>,
  'b': array([2.00247277e+00, 1.13385760e+00, 7.30052115e-04, 1.00000000e+00]),
  'c': array([-1.,  0.,  0.]),
  'dims': (zero: 4, nonneg: 2, exp: 0, soc: [], psd: []),
  'h': array([0]),
  'param_prob': <cvxpy.reductions.dcp2cone.cone_matrix_stuffing.ParamConeProg object at 0x12e286410>},
 Chain(reductions=[<cvxpy.reductions.flip_objective.FlipObjective object at 0x113663ed0>, <cvxpy.reductions.dcp2cone.dcp2cone.Dcp2Cone object at 0x12e253650>, <cvxpy.reductions.cvx_attr2constr.CvxAttr2Constr object at 0x12e2536d0>, <cvxpy.reductions.dcp2cone.cone_matrix_stuffing.ConeMatrixStuffing object at 0x12e253910>, <cvxpy.reductions.solvers.conic_solvers.cvxopt_conif.CVXOPT object at 0x12d718450>]),
 [[],
  <cvxpy.reductions.inverse_data.InverseData object at 0x12e253850>,
  ({0: Variable((2,)), 1: Variable(())},
   {0: Variable((2,), nonneg=True), 1: Variable(())},
   {10: 10, 15: 15}),
  <cvxpy.reductions.inverse_data.InverseData object at 0x112824850>,
  {'dims': (zero: 4, nonneg: 2, exp: 0, soc: [], psd: []),
   'eq_constr': [Zero(Expression(AFFINE, UNKNOWN, (3,))),
                 Zero(Expression(AFFINE, UNKNOWN, ()))],
   'offset': array(0.),
   'other_constr': [NonNeg(Expression(AFFINE, UNKNOWN, (2,)))],
   'var_id': 47}])
```

**Comment by rileyjmurray** [comment]
The history for our CVXOPT interface is here:
https://github.com/cvxgrp/cvxpy/commits/master/cvxpy/reductions/solvers/conic_solvers/cvxopt_conif.py
I believe 1.1.6 is hitting the code-path introduced in this commit:
https://github.com/cvxgrp/cvxpy/commit/c191e95efff98aa0eb86f7468923a2d394068692#diff-374cacac9c2fc577cd07f48390b67f246ec4892666d3110188e2a02f4b58897fR123-R130.

**Comment by kmonson** [comment]
I'm seeing a similar problem. I manually applied the change in @rileyjmurray's https://github.com/cvxgrp/cvxpy/pull/1250 and it did not make any difference.

In my case the 1.0.31 produces a solution and 1.1.0 reports `cvxpy.error.SolverError: Problem is unbounded; no solution found`. (It produces the same error with any of the 1.1.* versions.)

It looks like it's generating radically different problems for the solver when I change versions.

A and G are different dimensions, dims is different, h is different, and the inverse data is very different (only start shown). c is identical.

Version 1.0.31:
```
[{'A': <8761x35040 sparse matrix of type '<class 'numpy.float64'>'
	with 35040 stored elements in Compressed Sparse Column format>,
  'G': <123005x35040 sparse matrix of type '<class 'numpy.float64'>'
	with 175198 stored elements in Compressed Sparse Column format>,
  'b': array([-0., -0., -0., ..., -0., -0., -0.]),
  'c': array([*REDACTED*, ...,   0.  ,   0.  ,   0.  ]),
  'dims': (zero: 8761, nonpos: 123005, exp: 0, soc: [], psd: []),
  'h': array([ -0.,  -0.,  -0., ..., 800., 800., 800.]),
  'offset': array([0.])},
 Chain(reductions=[<cvxpy.reductions.dcp2cone.dcp2cone.Dcp2Cone object at 0x12c3953d0>, <cvxpy.reductions.cvx_attr2constr.CvxAttr2Constr object at 0x12c395580>, <cvxpy.reductions.eval_params.EvalParams object at 0x12c395c70>, <cvxpy.reductions.dcp2cone.cone_matrix_stuffing.ConeMatrixStuffing object at 0x12c3917f0>, <cvxpy.reductions.solvers.conic_solvers.ecos_conif.ECOS object at 0x12b0c8910>]),
 [<cvxpy.reductions.inverse_data.InverseData object at 0x12c391190>,
  [{0: ('var7186', Variable((8760,)), None),
    1: ('ene', Variable((8760,)), None),
    2: ('dis', Variable((8760,)), None),
    3: ('ch', Variable((8760,)), None)},
   {0: ('pv_out', Variable((8760,), nonneg=True), None),
    1: ('ene', Variable((8760,)), None),
    2: ('dis', Variable((8760,)), None),
    3: ('ch', Variable((8760,)), None)},
   {30: 30,
    34: 34,
    38: 38,
```

Version 1.1.0:
```
[{'A': <8761x43800 sparse matrix of type '<class 'numpy.float64'>'
	with 35040 stored elements in Compressed Sparse Column format>,
  'G': <26280x43800 sparse matrix of type '<class 'numpy.float64'>'
	with 26280 stored elements in Compressed Sparse Column format>,
  'b': array([0., 0., 0., ..., 0., 0., 0.]),
  'c': array([*REDACTED*, ...,   0.  ,   0.  ,   0.  ]),
  'dims': (zero: 8761, nonneg: 26280, exp: 0, soc: [], psd: []),
  'h': array([0., 0., 0., ..., 0., 0., 0.]),
  'param_prob': <cvxpy.reductions.dcp2cone.cone_matrix_stuffing.ParamConeProg object at 0x122d3a880>},
 Chain(reductions=[<cvxpy.reductions.dcp2cone.dcp2cone.Dcp2Cone object at 0x121e152b0>, <cvxpy.reductions.cvx_attr2constr.CvxAttr2Constr object at 0x121e154f0>, <cvxpy.reductions.eval_params.EvalParams object at 0x121e0f160>, <cvxpy.reductions.dcp2cone.cone_matrix_stuffing.ConeMatrixStuffing object at 0x121e0f850>, <cvxpy.reductions.solvers.conic_solvers.ecos_conif.ECOS object at 0x120e07ca0>]),
 [<cvxpy.reductions.inverse_data.InverseData object at 0x121e0b340>,
  [{0: ('var0', Variable((8760,)), None),
    1: ('ene', Variable((8760,)), None),
    2: ('dis', Variable((8760,)), None),
    3: ('ch', Variable((8760,)), None),
    4299: ('var4299', Variable((24,)), None),
    4319: ('var4319', Variable((24,)), None),
    4339: ('var4339', Variable((24,)), None),
    4359: ('var4359', Variable((24,)), None),
    4379: ('var4379', Variable((24,)), None),
    4399: ('var4399', Variable((24,)), None),
```

**Comment by rileyjmurray** [comment]
@kmonson do other solvers exhibit common behavior between CVXPY 1.0 and 1.1? For example, if you call your problem with ECOS in CVXPY 1.1, do you get the same result as in CVXPY 1.0?

**Comment by kmonson** [comment]
I've been running with ECOS. I haven't tested all of them but I know that SCS displays similar behavior. 

SCS Version 1.0.31:
```
[{'A': <131766x35040 sparse matrix of type '<class 'numpy.float64'>'
	with 210238 stored elements in Compressed Sparse Column format>,
  'b': array([ -0.,  -0.,  -0., ..., 800., 800., 800.]),
  'c': array([*CUT*, ...,   0.  ,   0.  ,   0.  ]),
  'dims': (zero: 8761, nonpos: 123005, exp: 0, soc: [], psd: []),
  'offset': array([0.])},
 Chain(reductions=[<cvxpy.reductions.dcp2cone.dcp2cone.Dcp2Cone object at 0x12df69430>, <cvxpy.reductions.cvx_attr2constr.CvxAttr2Constr object at 0x12df695e0>, <cvxpy.reductions.eval_params.EvalParams object at 0x12df69cd0>, <cvxpy.reductions.dcp2cone.cone_matrix_stuffing.ConeMatrixStuffing object at 0x12df65850>, <cvxpy.reductions.solvers.conic_solvers.scs_conif.SCS object at 0x12cd266a0>]),
 [<cvxpy.reductions.inverse_data.InverseData object at 0x12df651f0>,
  [{0: ('var7186', Variable((8760,)), None),
    1: ('ene', Variable((8760,)), None),
    2: ('dis', Variable((8760,)), None),
    3: ('ch', Variable((8760,)), None)},
   {0: ('pv_out', Variable((8760,), nonneg=True), None),
    1: ('ene', Variable((8760,)), None),
    2: ('dis', Variable((8760,)), None),
    3: ('ch', Variable((8760,)), None)},
   {30: 30,
    34: 34,
```

SCS Version 1.1.0:
```
[{'A': <35041x43800 sparse matrix of type '<class 'numpy.float64'>'
	with 61320 stored elements in Compressed Sparse Column format>,
  'b': array([0., 0., 0., ..., 0., 0., 0.]),
  'c': array([-36.73, -36.47, -36.57, ...,   0.  ,   0.  ,   0.  ]),
  'dims': (zero: 8761, nonneg: 26280, exp: 0, soc: [], psd: []),
  'param_prob': <cvxpy.reductions.dcp2cone.cone_matrix_stuffing.ParamConeProg object at 0x1243ad9a0>},
 Chain(reductions=[<cvxpy.reductions.dcp2cone.dcp2cone.Dcp2Cone object at 0x1234873d0>, <cvxpy.reductions.cvx_attr2constr.CvxAttr2Constr object at 0x123487610>, <cvxpy.reductions.eval_params.EvalParams object at 0x123482160>, <cvxpy.reductions.dcp2cone.cone_matrix_stuffing.ConeMatrixStuffing object at 0x123482970>, <cvxpy.reductions.solvers.conic_solvers.scs_conif.SCS object at 0x12247cb50>]),
 [<cvxpy.reductions.inverse_data.InverseData object at 0x123482e20>,
  [{0: ('var0', Variable((8760,)), None),
    1: ('ene', Variable((8760,)), None),
    2: ('dis', Variable((8760,)), None),
    3: ('ch', Variable((8760,)), None),
    4299: ('var4299', Variable((24,)), None),
    4319: ('var4319', Variable((24,)), None),
    4339: ('var4339', Variable((24,)), None),
    4359: ('var4359', Variable((24,)), None),
    4379: ('var4379', Variable((24,)), None),
    4399: ('var4399', Variable((24,)), None),
```

**Comment by kmonson** [comment]
@rileyjmurray Do you want me to test any of the others?

**Comment by rileyjmurray** [comment]
@kmonson no need to test solvers other than SCS / ECOS. It's clear that the data generated for SCS is also very different between 1.0 and 1.1. To clarify: do the final solutions (the ones visible to the user) change when using SCS in CVXPY 1.0/1.1? If the solutions do change in that case, then we should discuss this in a separate GitHub issue.

**Comment by kmonson** [comment]
I'm not sure about SCS giving a different answer in 1.0 per se. SCS takes so long to compete in 1.0 that I usually kill it. In 1.1 it returns right away with an unbounded error just like ECOS. I'll fire it up again and wait it out to see what happens.

**Comment by rileyjmurray** [comment]
@kmonson making the comparison between 1.0/1.1 with ECOS is also fine. I'm just trying to isolate whether the discrepancy you're seeing is really related to CVXOPT.

**Comment by kmonson** [comment]
@rileyjmurray 

ECOS gives us a viable answer in 1.0 and has a status of unbounded in 1.1 (and a garbage answer, of course). I keep the same version of ECOS between tests.

**Comment by rileyjmurray** [comment]
Okay, that makes this an issue about something beyond CVXOPT. Can you please create a new GitHub issue and summarize what we've discussed on this topic, up to now? We'll be sure to attend to that soon.

**Comment by kmonson** [comment]
@rileyjmurray Well crap, I misread CVXOPT as CVXPY. I'll create a new ticket right away.

**Comment by rileyjmurray** [comment]
@kmonson no worries, that happens sometimes!

**Comment by rileyjmurray** [comment]
This was resolved by #1427. Thanks, @phschiele!
